pyomo/solvers/plugins/solvers/GAMS.py

```diff
@@ -287,20 +287,20 @@ def solve(self, *args, **kwds):
             finally:
                 # Always name working directory or delete files,
                 # regardless of any errors.
-                if keepfiles:
+                if keepfiles and tee:
```

Comment:
It seemed appropriate to me. I don't mind using any other option instead of `tee`. Any suggestions? The only thing that mattered to me was to have the possibility to fully disable output.

Reply:
I would advocate removing the dependence on `tee`, and instead have all solver output be sent out through a logger with either `pyomo.solvers` or `__name__` scoping.  

(One of my longstanding TODOs is to replace all `print()` and `sys.stdout.write()` calls with appropriate logger calls... the solver interfaces are some of the worst offenders)

Reply:
I undid my changes here and leave the printing issue for you. I will simply disable those lines locally for me then.